OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -153,4 +148,37 @@ void MarkersReference::setMarkerWeightSet(const Set<MarkerWeight>& markerWeights
     upd_marker_weights() = markerWeights;
 }
 
+void MarkersReference::setDefaultWeight(double weight)
+{
+    set_default_weight(weight); 
+}
+
+void MarkersReference::updateInternalWeights() const
```

Comment:
The `MarkersReference` maintains a flat private list of all the marker weights in the same order as the marker values.  This was done for convenience since the weights come from either the marker weight set or default weight properties, which are generally not modified and so there is efficiency to be gained by maintaining the internal `_weights`. If the weight properties are updated the output weights (`getWeights()`) must make sure the list is up-to-date. To ensure weights are always up to date with property values,  the update is called within the `getWeights() const` and therefore must also be a const method. The `updateInternalWeights() const` does not change any of the properties of the `MarkersReference` it merely transcribes the property values (const) into a single list for convenience/efficiency.